app/src/main/java/com/yacgroup/yacguide/CountryActivity.kt

```diff
@@ -67,4 +78,26 @@ class CountryActivity : UpdatableTableActivity() {
     override fun getLayoutId(): Int {
         return R.layout.activity_country
     }
+
+    /*
+     * Check whether the "What's New" dialog is shown or not. This is decided by comparing
+     * the version stored in the settings with the current app version.
+     */
+    private fun _init() {
```

Comment:
This method has nothing to do with Country selection. I would move it the the `WhatsNew` class and call sth like `checkForVersionUpdate()`. In the constructor of this activity, you could then do:
```
val whatsNewInfo = WhatsNew(this)
if (whatsNewInfo.checkForVersionUpdate()) {
whatsNewInfo.showDialog()
} 
```

The `SharedPreferences` can be moved to WhatsNew class, too.